networkit/cpp/community/GraphClusteringTools.cpp

```diff
@@ -14,8 +14,17 @@ float getImbalance(const Partition &zeta) {
     std::vector<count> clusterSizes = zeta.subsetSizes();
     float maxClusterSize = (float) *std::max_element(clusterSizes.begin(),
             clusterSizes.end());
-    float imbalance = maxClusterSize / avg;
-    return imbalance;
+    return static_cast<float>(maxClusterSize / avg);
+}
+
+float getImbalance(const Partition &zeta, const Graph &graph) {
+    float avg = std::ceil(
+        (float)graph.numberOfNodes()
+        / (float)zeta.numberOfSubsets());
+    std::vector<count> clusterSizes = zeta.subsetSizes();
+    float maxClusterSize = (float) *std::max_element(clusterSizes.begin(),
+            clusterSizes.end());
+    return static_cast<float>(maxClusterSize / avg);
```

Comment:
Both variables are float, the cast should not be necessary
```suggestion
    return maxClusterSize / avg;
```

Reply:
The two functions are very similar, the only difference is the computation of `avg`. I suggest having a single private function (e.g., `computeImbalance`) and using it in the two public `getImbalace`.